src/radical/pilot/agent/staging_input/default.py

```diff
@@ -176,29 +177,41 @@ def _handle_unit(self, unit, actionables):
             if rpc.CREATE_PARENTS in flags and action != rpc.TRANSFER:
                 tgtdir = os.path.dirname(tgt.path)
                 if tgtdir != sandbox:
-                    # TODO: optimization point: create each dir only once
-                    self._log.debug("mkdir %s", tgtdir)
-                    rpu.rec_makedir(tgtdir)
-
-            if   action == rpc.COPY: shutil.copyfile(src.path, tgt.path)
+                    # create each dir only once
+                    if not os.path.isdir(tgtdir):
+                        self._log.debug("mkdir %s", tgtdir)
+                        rpu.rec_makedir(tgtdir)
```

Comment:
`rpu.rec_makeddirs` (calls)[https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/utils/misc.py#L53) `os.makedirs` , the first action of that is to (chec)[https://github.com/python/cpython/blob/master/Lib/os.py#L209] if the dir exists.  In this case the comment was out of sync with the implementation, which shows, again, that incorrect comments are worse than no comments...  Apologies!